modules/heat_conduction/include/userobjects/ViewFactorBase.h

```diff
@@ -0,0 +1,67 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "SideUserObject.h"
+
+// Forward Declarations
+class ViewFactorBase;
+
+template <>
+InputParameters validParams<ViewFactorBase>();
+
+/**
+ * A base class for automatic computation of view factors between sidesets
+ */
+class ViewFactorBase : public SideUserObject
+{
+public:
+  ViewFactorBase(const InputParameters & parameters);
+
+  ///@{ public interface for obtaining view factors
+  Real getViewFactor(BoundaryID from_id, BoundaryID to_id) const;
+  Real getViewFactor(BoundaryName from_name, BoundaryName to_name) const;
```

Comment:
I'm not crazy about keeping track of Names and IDs everywhere. In MOOSE we convert names and IDs or even combinations of them all to IDs and then use IDs everywhere. It seems like you have created extra bookkeeping by keeping both around.